client/src/components/Schedule/TableView.tsx

```diff
@@ -0,0 +1,165 @@
+          defaultSortOrder: 'descend',
+        },
+        { title: 'Time', width: 60, dataIndex: 'dateTime', render: timeRenderer(timeZone) },
+        {
+          title: 'Type',
+          width: 100,
+          dataIndex: ['event', 'type'],
+          render: (value: keyof typeof EventTypeColor) => (
+            <Tag color={EventTypeColor[value]}>{EventTypeToName[value] || value}</Tag>
+          ),
+        },
+        {
+          title: 'Special',
+          width: 150,
+          dataIndex: 'special',
+          render: tags => {
+            if (!tags) {
+              return;
+            }
+            return tagsRenderer(tags.split(','));
+          },
+        },
+        {
+          title: 'Name',
+          dataIndex: ['event', 'name'],
+          render: (value: string, record) => {
+            return record.event.descriptionUrl ? (
+              <a target="_blank" href={record.event.descriptionUrl}>
+                {value}
+              </a>
+            ) : (
+              value
+            );
+          },
+          ...getColumnSearchProps('event.name'),
+        },
+        {
+          title: 'Url',
+          width: 30,
+          dataIndex: 'broadcastUrl',
+          render: url => {
+            return url ? (
+              <Tooltip placement="topLeft" title={url}>
+                <a target="_blank" href={url}>
+                  {urlRenderer(url)}
+                </a>
+              </Tooltip>
+            ) : (
+              ''
+            );
+          },
+        },
+        { title: 'Duration', width: 60, dataIndex: 'duration' },
+        {
+          title: 'Organizer',
+          width: 140,
+          dataIndex: ['organizer', 'githubId'],
+          render: (value: string) => (value ? <GithubUserLink value={value} /> : ''),
+          ...getColumnSearchProps('organizer.githubId'),
+        },
+        {
+          title: 'Place',
+          dataIndex: 'place',
+          render: (value: string) => {
+            return value === 'Youtube Live' ? (
+              <div>
+                <YoutubeOutlined /> {value}{' '}
+                <Tooltip title="Ссылка будет в Discord">
+                  <QuestionCircleOutlined />
+                </Tooltip>
+              </div>
+            ) : (
+              <Tooltip title={value}>
+                {<div style={{ textOverflow: 'ellipsis', overflow: 'hidden' }}>{value}</div>}
+              </Tooltip>
+            );
+          },
+          onCell: () => {
+            return {
+              style: {
+                whiteSpace: 'nowrap',
+                maxWidth: 250,
+              },
+            };
+          },
+        },
+      ]}
+    />
+  );
+}
+
+const dateRenderer = (timeZone: string) => (value: string) =>
+  value ? moment(value, 'YYYY-MM-DD HH:mmZ').tz(timeZone).format('MMM Do YYYY') : '';
+
+const timeRenderer = (timeZone: string) => (value: string) =>
+  value ? moment(value, 'YYYY-MM-DD HH:mmZ').tz(timeZone).format('HH:mm') : '';
+
```

Comment:
Let's replace these two functions with a more generic one:

```
const renderDate = (timeZone: string, format: string) => (value: string) =>
  value ? moment(value, 'YYYY-MM-DD HH:mmZ').tz(timeZone).format(format) : '';
```

Reply:
or consider some of the methods from `client/src/components/Table/renderers.tsx`

Reply:
done